client/src/components/content/CharacterShow.js

```diff
@@ -1,37 +1,75 @@
 import React, { useState, useEffect } from "react"
+import { useParams } from "react-router-dom";
+import VoteButtons from "./VoteButtons.js"
 
 const CharacterShow = (props) => {
   const [character, setCharacter] = useState({})
-  
+  const [totalVotes, setTotalVotes] = useState(0)
+  const [userVote, setUserVote] = useState(false)
+  const [clicked, setClicked] = useState(0)
+  const params = useParams()
+
   const getCharacter = async () => {
-    const characterId = props.match.params.id
+    const characterId = params.id
     try {
       const response = await fetch(`/api/v1/characters/${characterId}`)
       if (!response.ok) {
         const errorMessage = `${response.status} (${response.statusText})`
         const error = new Error(errorMessage)
-        throw(error)
+        throw (error)
       }
       const body = await response.json()
       setCharacter(body.character)
-    }catch(error) {
+      setTotalVotes(body.character.votes[0].sum)
```

Comment:
Let's perform total calculations on the backend when we are gathering all the data so that the front end here is just responsible for displaying said data